platform/ios/src/MGLAPIClient.m

```diff
@@ -88,7 +88,7 @@ - (NSURLRequest *)requestForEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)ev
 
 - (void)setupBaseURL {
     NSString *testServerURL = [[NSUserDefaults standardUserDefaults] stringForKey:@"MGLTelemetryTestServerURL"];
-    if (testServerURL) {
+    if (testServerURL && [testServerURL hasPrefix:@"https://"]) {
```

Comment:
We should check whether the URL is well-formed, by checking whether an NSURL can be initialized from it. Also, with the NSURL, we can compare its `scheme` property.